source/frontend/src/features/mapSideBar/disposition/tabs/offersAndSale/dispositionSale/update/UpdateDispostionSaleView.tsx

```diff
@@ -0,0 +1,121 @@
+  loading: boolean;
+  onSave: (sale: Api_DispositionFileSale) => Promise<Api_DispositionFileSale | undefined>;
+  onCancel: () => void;
+  onSuccess: () => void;
+  onError: (e: AxiosError<IApiError>) => void;
+}
+
+const UpdateDispositionSaleView: React.FC<IUpdateDispositionSaleViewProps> = ({
+  initialValues,
+  loading,
+  onSave,
+  onCancel,
+  onSuccess,
+  onError,
+}) => {
+  const { setModalContent, setDisplayModal } = useModalContext();
+
+  const cancelFunc = (resetForm: () => void, dirty: boolean) => {
+    if (!dirty) {
+      resetForm();
+      onCancel();
+    } else {
+      setModalContent({
+        ...getCancelModalProps(),
+        handleOk: () => {
+          resetForm();
+          setDisplayModal(false);
+          onCancel();
+        },
+      });
+      setDisplayModal(true);
+    }
+  };
+
+  return (
+    <StyledFormWrapper>
+      <Formik<DispositionSaleFormModel>
+        enableReinitialize
+        validationSchema={DispositionSaleFormYupSchema}
+        initialValues={initialValues}
+        onSubmit={async (values: DispositionSaleFormModel, formikHelpers) => {
+          try {
+            const sale = await onSave(values.toApi());
+            if (sale && sale.id) {
+              onSuccess();
+            }
+          } catch (e) {
+            if (axios.isAxiosError(e)) {
+              const axiosError = e as AxiosError<IApiError>;
+              onError && onError(axiosError);
+            }
+          } finally {
+            formikHelpers.setSubmitting(false);
+          }
+        }}
+      >
+        {formikProps => {
+          return (
+            <>
+              <LoadingBackdrop
+                show={formikProps.isSubmitting || loading}
+                parentScreen={true}
+              ></LoadingBackdrop>
+              <StyledContent>
+                <DispositionSaleForm dispostionSaleId={initialValues.id} />
+              </StyledContent>
+              <StyledFooter>
+                <SidebarFooter
+                  onSave={() => formikProps.submitForm()}
+                  isOkDisabled={formikProps.isSubmitting || !formikProps.dirty}
+                  onCancel={() => cancelFunc(formikProps.resetForm, formikProps.dirty)}
+                  displayRequiredFieldError={
+                    formikProps.isValid === false && !!formikProps.submitCount
+                  }
+                />
+              </StyledFooter>
+            </>
+          );
+        }}
+      </Formik>
+    </StyledFormWrapper>
+  );
+};
+
+export default UpdateDispositionSaleView;
+
+const StyledFormWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  flex-grow: 1;
+  text-align: left;
+  height: 100%;
+  overflow-y: auto;
+  padding-bottom: 1rem;
+`;
+
+const StyledContent = styled.div`
+  background-color: ${props => props.theme.css.filterBackgroundColor};
+`;
+
+const StyledFooter = styled.div`
+  margin-right: 1rem;
+  padding-bottom: 1rem;
+  z-index: 0;
+`;
```

Comment:
any chance this is duplicated?

Reply:
Very likely, we could create a story to define a new css definition for these kind of components.